File: frontend/src/types/product.ts
export type Product = {
  id: string
  slug: string
  name: string
  category: string
  price: number
  sizes: string[]
  image: string
  tone: string
  isNew?: boolean
  variants?: Array<{ id: string; sku: string; size: string; stockQuantity: number }>
}

export type CartItem = Product & { size: string; quantity: number }

File: services/catalog-service/app/schemas/product.py
from uuid import UUID

from pydantic import BaseModel, Field


class ProductVariantResponse(BaseModel):
    id: UUID
    sku: str
    size: str
    stock_quantity: int = Field(ge=0)


class ProductPreview(BaseModel):
    id: UUID
    slug: str
    name: str
    description: str
    category: str
    price_kopecks: int = Field(ge=0, description="Цена в копейках")
    color: str
    material: str
    tone: str
    sizes: list[str]
    variants: list[ProductVariantResponse]
    image: str
    is_new: bool = False
